src/main/java/io/cdap/plugin/ariba/source/config/AribaPluginConfig.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.ariba.source.config;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.ariba.source.util.AribaUtil;
+import io.cdap.plugin.ariba.source.util.ResourceConstants;
+import io.cdap.plugin.common.IdUtils;
+import io.cdap.plugin.common.ReferencePluginConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+/**
+ * This {@code AribaPluginConfig} contains all Ariba configuration parameters.
+ */
+
+public class AribaPluginConfig extends ReferencePluginConfig {
+
+  public static final String BASE_URL = "baseURL";
+  public static final String SYSTEM_TYPE = "systemType";
+  public static final String REALM = "realm";
+  public static final String TEMPLATE_NAME = "viewTemplateName";
+  public static final String CLIENT_ID = "clientId";
+  public static final String CLIENT_SECRET = "clientSecret";
+  public static final String APIKEY = "apiKey";
+  public static final String FROM_DATE = "fromDate";
+  public static final String TO_DATE = "toDate";
+
+  private static final Logger LOG = LoggerFactory.getLogger(AribaPluginConfig.class);
+  private static final String COMMON_ACTION = ResourceConstants.ERR_MISSING_PARAM_OR_MACRO_ACTION.getMsgForKey();
+  /**
+   * Basic parameters.
+   */
+  @Macro
+  @Description("Base Path of Ariba API.")
+  private final String baseURL;
+
+  @Description("Select System either prod/sandbox.")
+  private final String systemType;
```

Comment:
Yes, it only accepts two words either 'prod' or 'sandbox' and in 99% case user need to select 'prod' as in case of 'sandbox' Ariba sends some mocked results for every API call so it can only be used for some tests when user don't have actual authorisation. And to avoid user mistake this has been marked non-macro field.